src/Microsoft.ML.Transforms/Text/StopWordsRemoverTransform.cs

```diff
@@ -286,121 +187,44 @@ private static NormStr.Pool[] StopWords
-                CheckResources(ch);
+                Input = input;
+                Output = output;
+                Language = language;
+                LanguageColumn = languageColumn;
             }
-            Metadata.Seal();
         }
 
-        private void CheckResources(IChannel ch)
+        private static (string input, string output)[] GetColumnPairs(ColumnInfo[] columns)
         {
-            Host.AssertValue(ch);
-
-            // Find required resources
-            var requiredResources = new bool[StopWords.Length];
-            for (int iinfo = 0; iinfo < Infos.Length; iinfo++)
-                requiredResources[(int)_exes[iinfo].Lang] = true;
-
-            // Check the existence of resource files
-            var missings = new StringBuilder();
-            foreach (Language lang in Enum.GetValues(typeof(Language)))
-            {
-                if (GetResourceFileStreamOrNull(lang) == null)
-                {
-                    if (requiredResources[(int)lang])
-                    {
-                        throw ch.Except(
-                            "Missing '{0}.txt' resource.");
-                    }
-
-                    if (missings.Length > 0)
-                        missings.Append(", ");
-                    missings.Append(lang);
-                }
-            }
-
-            if (missings.Length > 0)
-            {
-                const string wrnMsg = "Missing resources for languages: '{0}'. You can check the following help page for more info: '{1}'. "
-                    + "Default stop words list (specified by 'lang' option) will be used if needed.";
-                ch.Warning(wrnMsg, missings.ToString());
-            }
+            Contracts.CheckValue(columns, nameof(columns));
+            return columns.Select(x => (x.Input, x.Output)).ToArray();
         }
 
-        public static StopWordsRemoverTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        /// <summary>
+        /// Stopword remover removes language-specific lists of stop words (most common words).
+        /// </summary>
+        /// <param name="env">The environment.</param>
+        /// <param name="columns">Pairs of columns to remove stop words from.</param>
+        public StopWordsRemoverTransform(IHostEnvironment env, params ColumnInfo[] columns) :
+            base(Contracts.CheckRef(env, nameof(env)).Register(RegistrationName), GetColumnPairs(columns))
         {
-            Contracts.CheckValue(env, nameof(env));
-            env.CheckValue(ctx, nameof(ctx));
-            ctx.CheckAtModel(GetVersionInfo());
-            env.CheckValue(env, nameof(env));
-            env.CheckValue(input, nameof(input));
-
-            var h = env.Register(RegistrationName);
-            return h.Apply("Loading Model", ch => new StopWordsRemoverTransform(h, ctx, input));
+            _columns = columns;
         }
 
         public override void Save(ModelSaveContext ctx)
```

Comment:
>publi [](start = 8, length = 5)

doc

Reply:
It's an override, right? I thought the public overrides don't need to be documented (because the base methods would typically have comments, and they are exactly the same)

---
In reply to: [227844134](https://github.com/dotnet/machinelearning/pull/1338#discussion_r227844134) [](ancestors = 227844134)